### ISSUE TITLE
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module

### ISSUE THREAD

**Issue by MichaelGradek** [sindresorhus/file-type#535]
I just installed the library onto my TypeScript project and I get the following error:

Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /Users/mike/project/node_modules/file-type/index.js
require() of ES modules is not supported.

Any idea why this may be happening?

"file-type": "^17.1.1"

**Comment by sindresorhus** [comment]
Duplicate of #525

**Comment by Borewit** [comment]
To import a ES module from a CommonJs module, is very tricky.
The `import` is transpiled to `require`, that is why you get the error `require() of ES modules is not supported`.

Have look to this artice [When a CommonJS module imports another CommonJS module](https://javascript.plainenglish.io/how-to-correctly-use-typescript-module-import-syntax-and-settings-in-various-circumstances-e98bfa87f70f), check _When an ES Module imports a CommonJS module_.

Or use a quick and dirty workaround:
```ts
/**
 * Import 'file-type' ES-Module in CommonJS Node.js module
 */
(async () => {
  const { fileTypeFromFile } = await (eval('import("file-type")') as Promise<typeof import('file-type')>);

  const type = await fileTypeFromFile('fixture.skp');
  console.log(type);
})();
```

**Comment by Luxcium** [comment]
```log
❯ ts-node "${PROJECT}/src/utilities/sharp-phash/__sharp-phash-working.ts"
${PROJECT}/src/packages/file-path/getPathWithStats.ts:4
import { FileTypes } from './tools';
                    ^
Error [ERR_REQUIRE_ESM]: require() of ES Module ${PROJECT}/node_modules/file-type/index.js from ${PROJECT}/src/packages/file-path/getPathWithStats.ts not supported.
Instead change the require of index.js in ${PROJECT}/src/packages/file-path/getPathWithStats.ts to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (${PROJECT}/src/packages/file-path/getPathWithStats.ts:4:21)
    at Module.m._compile (${HOME}/.nvm/versions/node/v17.5.0/lib/node_modules/ts-node/dist/index.js:786:29)
    at Object.require.extensions.<computed> [as .ts] (${HOME}/.nvm/versions/node/v17.5.0/lib/node_modules/ts-node/dist/index.js:788:16)
    at Object.<anonymous> (${PROJECT}/src/packages/file-path/utils/index.ts:4:26)
    at Module.m._compile (${HOME}/.nvm/versions/node/v17.5.0/lib/node_modules/ts-node/dist/index.js:786:29)
    at Object.require.extensions.<computed> [as .ts] (${HOME}/.nvm/versions/node/v17.5.0/lib/node_modules/ts-node/dist/index.js:788:16)
    at Object.<anonymous> (${PROJECT}/src/packages/file-path/tools/dirListWithFileType/currentPath.ts:4:17)
    at Module.m._compile (${HOME}/.nvm/versions/node/v17.5.0/lib/node_modules/ts-node/dist/index.js:786:29)
    at Object.require.extensions.<computed> [as .ts] (${HOME}/.nvm/versions/node/v17.5.0/lib/node_modules/ts-node/dist/index.js:788:16)
    at Object.<anonymous> (${PROJECT}/src/packages/file-path/tools/dirListWithFileType/index.ts:4:21)
    at Module.m._compile (${HOME}/.nvm/versions/node/v17.5.0/lib/node_modules/ts-node/dist/index.js:786:29)
    at Object.require.extensions.<computed> [as .ts] (${HOME}/.nvm/versions/node/v17.5.0/lib/node_modules/ts-node/dist/index.js:788:16)
    at Object.<anonymous> (${PROJECT}/src/utilities/sharp-phash/__sharp-phash-working.ts:7:31)
    at Module.m._compile (${HOME}/.nvm/versions/node/v17.5.0/lib/node_modules/ts-node/dist/index.js:786:29)
    at Object.require.extensions.<computed> [as .ts] (${HOME}/.nvm/versions/node/v17.5.0/lib/node_modules/ts-node/dist/index.js:788:16)
    at main (${HOME}/.nvm/versions/node/v17.5.0/lib/node_modules/ts-node/dist/bin.js:292:16)
    at Object.<anonymous> (${HOME}/.nvm/versions/node/v17.5.0/lib/node_modules/ts-node/dist/bin.js:405:5) {
  code: 'ERR_REQUIRE_ESM'
}

```

**Comment by Luxcium** [comment]
and when using the compiled version of the code:

```log
❯ node "/home/luxcium/projects/parallel-mapping/out/src/utilities/sharp-phash/__sharp-phash-working.js"
/home/luxcium/projects/parallel-mapping/out/src/packages/file-path/getPathWithStats.js:4
const file_type_1 = require("file-type");
                    ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /home/luxcium/projects/parallel-mapping/node_modules/file-type/index.js from /home/luxcium/projects/parallel-mapping/out/src/packages/file-path/getPathWithStats.js not supported.
Instead change the require of index.js in /home/luxcium/projects/parallel-mapping/out/src/packages/file-path/getPathWithStats.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/home/luxcium/projects/parallel-mapping/out/src/packages/file-path/getPathWithStats.js:4:21)
    at Object.<anonymous> (/home/luxcium/projects/parallel-mapping/out/src/packages/file-path/utils/index.js:4:26)
    at Object.<anonymous> (/home/luxcium/projects/parallel-mapping/out/src/packages/file-path/tools/dirListWithFileType/currentPath.js:4:17)
    at Object.<anonymous> (/home/luxcium/projects/parallel-mapping/out/src/packages/file-path/tools/dirListWithFileType/index.js:4:21)
    at Object.<anonymous> (/home/luxcium/projects/parallel-mapping/out/src/utilities/sharp-phash/__sharp-phash-working.js:7:31) {
  code: 'ERR_REQUIRE_ESM'
}

Node.js v17.7.1


```

**Comment by Luxcium** [comment]
I can not do any of the different methods I am aware of:

![Screenshot_000001_20220315_200858](https://user-images.githubusercontent.com/42672814/158492121-e5ca925f-95a2-4b89-8336-4fc89f91e1a1.png)

**Comment by azhengyongqin** [comment]
Stupid updates

**Comment by kkan0615** [comment]
It makes me super annoying!!!!!!

**Comment by Thomas-1985** [comment]
I really want to use your npm in my TypeScript project and tried import and require, none of them seem to work. The error message is
`
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /my-project/index.js
require() of ES modules is not supported.
require() of /my-project/node_modules/file-type/index.js from /my-project/users.ts is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /my-project/node_modules/file-type/package.json.

    at new NodeError (internal/errors.js:322:7)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1102:13)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/my-project/users.ts:44:1)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Module.m._compile (/my-project/node_modules/ts-node/src/index.ts:1459:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:1114:10) {
  code: 'ERR_REQUIRE_ESM'
}
[nodemon] app crashed - waiting for file changes before starting...
`

Please make it so i can use the npm like with every other by using either import or require

**Comment by dedeardiansya** [comment]
hahahha, i like this update.

**Comment by dedeardiansya** [comment]
this makes me happy
`npm i file-type@16.5.3`

**Comment by swizes** [comment]
Why does a package update require a quick & dirty workaround

**Comment by netojose** [comment]
I had to make the same thing @dedeardiansya told, downgrade. 
I'm watching changes on this issue, when be solved, I update the package.

**Comment by jloehel** [comment]
> @netojose to which version do you downgrade?

The 17.0.0 is breaking everything. The versions before should work:
```
...
    "file-type": "^16.5.3",
...
```

**Comment by emimarz** [comment]
thanks

**Comment by sindresorhus** [comment]
TypeScript now fully supports ESM: https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#ecmascript-module-support-in-node-js

**Comment by prernagp** [comment]
This is still not working... have tried multiple things in my loopback project. I really need the latest fix but unable to do so

**Comment by T-vK** [comment]
Are there an plans to make file-type CommonJS-compatible again?

Are there any maintained alternatives out there that support getting the mime-type from a buffer like `FileType.fromBuffer`?

**Comment by prernagp** [comment]
@T-vK @vaibhavkumar-sf We have managed to use it like this:-

`const {fileTypeFromBuffer} = await (eval('import("file-type")') as Promise<typeof import('file-type')>);
const fileTrueType = await fileTypeFromBuffer(request);`

**Comment by vtereshyn** [comment]
> > @netojose to which version do you downgrade?
> 
> The 17.0.0 is breaking everything. The versions before should work:
> 
> ```
> ...
>     "file-type": "^16.5.3",
> ...
> ```

this version has an `audit` issue.

**Comment by netojose** [comment]
@vtereshyn , thanks for share this security information. So, we need to choose between the both options (make necessary changes to project to became full ESM, or accept the security issue :( )

**Comment by vtereshyn** [comment]
> @vtereshyn , thanks for share this security information. So, we need to choose between the both options (make necessary changes to project to became full ESM, or accept the security issue :( )

Yes. But I have huge project written in Typescript and I don't have an ability to change everything just now. That's why I am working on fork of this repo so I can use `commonjs`

**Comment by Borewit** [comment]
> or accept the security issue :( )

No need for that, that has been addressed in version [v16.5.4](https://github.com/sindresorhus/file-type/releases/tag/v16.5.4) @netojose.

**Comment by cheng-alvin** [comment]
> "file-type": "^16.5.3",
That's it! Thanks! 
Just a side note if you are importing 'fileTypeFromBuffer' (For determining type in v1.18) just modify it to 'fromBuffer'.
Otherwise, it will not compile as it is not featured in 'v1.16.x'

**Comment by nkemcels** [comment]
Downgrade to version 16
`yarn add file-type@16`

**Comment by breno-bicycle** [comment]
as mentioned before, the version 16.5.4 has audit issues: https://github.com/sindresorhus/file-type/issues/535#issuecomment-1240459304

It is a critical vulnerabilities that we don't want to have in our project.

So, the only alternative is still use it with `import()`?

should this issue be opened until a release supporting common import works?

**Comment by berenar** [comment]
My two cents here, upgrading to [yarn 3](https://yarnpkg.com/migration/guide) worked for me!

**Comment by soydev019** [comment]
I have the same issue, had to downgrade to v16.5.4 to make it work

**Comment by Shandur** [comment]
Thanks to @Borewit - it helped 🙌

**Comment by max-degterev** [comment]
Did anyone for this package already?
